Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,229 @@
+﻿using Data;
+using Data.Models.ProductTables;
+using Backend.Services;
+using Lib.EntityFrameworkCore;
+using Microsoft.AspNetCore.Mvc;
+using Lib.Storage;
+using Lib.Storage.Services;
+using System.Drawing;
+using System.Drawing.Imaging;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backend.Controllers
+{
+    [Route("v1/product")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        private readonly Db db;
+
+        private readonly ImageService imageService;
+
+        private readonly Storj storage;
+
+
+        public ProductController(Db context, ImageService imageService, Storj storage)
+        {
+            db = context;
+            this.imageService = imageService;
+            this.storage = storage;
+        }
+
+
+        [HttpPost]
+        public async Task<IActionResult> CreateProduct(Product newProduct)
+        {
+            newProduct.IsDraft = true;
+
+            db.Products.Add(newProduct);
+
+            var saved = await db.Save();
+
+            if (saved)
+            {
+                var createdProduct = await GetProduct(newProduct.Id);
+                if (createdProduct.Result is OkObjectResult okResult)
+                {
+                    return Ok(okResult.Value);
+                }
+            }
+
+            return Problem();
+        }
+
+    
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Product>> GetProduct(string id)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == id);
+
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(product);
+        }
+
+   
+        [HttpPatch("{id}")]
+        public async Task<IActionResult> UpdateProduct([FromBody] Product patchDoc)
```

Comment:
You should check for user. Add [Authorize]. Get uid by `User.FindFirst(Jwt.Uid)`. Check that shop is owned by user.